libyul/optimiser/DataFlowAnalyzer.cpp

```diff
@@ -112,10 +112,8 @@ void DataFlowAnalyzer::operator()(FunctionDefinition& _fun)
 
 void DataFlowAnalyzer::operator()(ForLoop& _for)
 {
-	// Special scope handling of the pre block.
-	pushScope(false);
```

Comment:
You removed the ``pushScope`` here, but still call ``popScope`` below - my guess is that that's the problem.
The fix is probably to just leave the ``pushScope(false);`` in here.